java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -167,16 +190,37 @@ private static Consumer createConsumer(
         consumer =  new AvroStringConsumer((VarCharVector) vector);
         break;
       case FIXED:
-        arrowType = new ArrowType.FixedSizeBinary(schema.getFixedSize());
-        fieldType =  new FieldType(nullable, arrowType, /*dictionary=*/null, getMetaData(schema));
-        vector = createVector(consumerVector, fieldType, name, allocator);
-        consumer =  new AvroFixedConsumer((FixedSizeBinaryVector) vector, schema.getFixedSize());
+        if (logicalType instanceof LogicalTypes.Decimal) {
+          final int scala = ((LogicalTypes.Decimal) logicalType).getScale();
+          final int precision = ((LogicalTypes.Decimal) logicalType).getPrecision();
+          arrowType = new ArrowType.Decimal(precision, scala);
```

Comment:
please validation precision and scale.  in particular scale can't be more then 38 for decimal128 i believe.